pkg/skaffold/build/cluster/pod.go

```diff
@@ -155,6 +148,18 @@ func (b *Builder) env(artifact *latest.KanikoArtifact, httpProxy, httpsProxy str
 		})
 	}
 
+	// if cluster.PullSecretName  is non-empty populate secret path and use as GOOGLE_APPLICATION_CREDENTIALS
```

Comment:
@chtcvl  Shouldn't both `PullSecretName` and `PullSecretPath` be empty when using workload identity?
In case if `pullSecretPath` is specified, skaffold will create a secret in the k8 cluster.

Reply:
@tejal29 they should be empty, but currently `pullSecretPath` is always non-empty - line 125 before change. This merge request addresses this issue - it populates `pullSecretPath` with ` strings.Join([]string{b.ClusterDetails.PullSecretMountPath, b.ClusterDetails.PullSecretPath}, "/", // linux filepath separator.` only if `PullSecretName` (which is a configuration setting) is not empty.
 		)`